tests/run/colltest4/CollectionStrawMan4_1.scala

```diff
@@ -217,7 +217,7 @@ object CollectionStrawMan4 {
     def fromIterable[B](coll: Iterable[B]): ListBuffer[B] = coll match {
       case pd @ View.Partitioned(partition: View.Partition[B]) =>
         partition.distribute(new ListBuffer[B]())
-        pd.forced.get.asInstanceOf[ListBuffer[B]]
+        new ListBuffer[B] ++= pd.forced.get.asInstanceOf[List[B]]
```

Comment:
You don't even need the cast, `pd.forced.get` works, but it would be interesting to know if this is what @odersky intented or if there's a design flaw somewhere. Also please squash your commits related to this into one commit